deps/mqtt-c/mqtt.c

```diff
@@ -1175,41 +1175,41 @@ ssize_t mqtt_pack_connection_request(uint8_t* buf, size_t bufsz,
     if (useTLS) {
         if (caFilePath != NULL) {
             /* a caFilePath is present */
-            connect_flags |= MQTT_CONNECT_CAFILEPATH;
+            connect_flags |= (uint8_t)~MQTT_CONNECT_CAFILEPATH;
             remaining_length += (uint32_t)__mqtt_packed_cstrlen(caFilePath);
         } else {
             connect_flags &= (uint8_t)~MQTT_CONNECT_CAFILEPATH;
         }
 
         if (caPath != NULL) {
             /* a caPath is present */
-            connect_flags |= MQTT_CONNECT_CAPATH;
+            connect_flags |= (uint8_t)~MQTT_CONNECT_CAPATH;
             remaining_length += (uint32_t)__mqtt_packed_cstrlen(caPath);
         } else {
             connect_flags &= (uint8_t)~MQTT_CONNECT_CAPATH;
         }
 
         if (clientCertPath != NULL) {
             /* a clientCertPath is present */
-            connect_flags |= MQTT_CONNECT_CLIENTCERTPATH;
+            connect_flags |= (uint8_t)~MQTT_CONNECT_CLIENTCERTPATH;
             remaining_length += (uint32_t)__mqtt_packed_cstrlen(clientCertPath);
         } else {
             connect_flags &= (uint8_t)~MQTT_CONNECT_CLIENTCERTPATH;
         }
 
         if (clientKeyPath != NULL) {
             /* a clientKeyPath is present */
-            connect_flags |= MQTT_CONNECT_CLIENTKEYPATH;
+            connect_flags |= (uint8_t)~MQTT_CONNECT_CLIENTKEYPATH;
             remaining_length += (uint32_t)__mqtt_packed_cstrlen(clientKeyPath);
         } else {
             connect_flags &= (uint8_t)~MQTT_CONNECT_CLIENTKEYPATH;
         }
 
         if (useTLS) /* useTLS = true */
-            strncpy(strTLS, "1", 1);
+            memcpy(strTLS, "1", 1);
         else
-            strncpy(strTLS, "0", 1);
-        connect_flags |= MQTT_CONNECT_USETLS;
+            memcpy(strTLS, "0", 1);
+        connect_flags |= (uint8_t)~MQTT_CONNECT_USETLS;
         remaining_length += (uint32_t)__mqtt_packed_cstrlen(strTLS);
```

Comment:
This line calls strlen.
You need to zero-terminate the strTLS value.

Reply:
I have changed the call of strncpy so that the strTLS value is zero-terminated.